crates/red_knot_ide/src/completion.rs

```diff
@@ -0,0 +1,43 @@
+use ruff_db::files::File;
+use ruff_db::parsed::parsed_module;
+use ruff_python_ast::visitor::source_order::{SourceOrderVisitor, TraversalSignal};
+use ruff_python_ast::AnyNodeRef;
+use ruff_text_size::TextSize;
+
+use crate::Db;
+
+pub struct Completion {
+    pub label: String,
+}
+
+pub fn completion(db: &dyn Db, file: File, _offset: TextSize) -> Vec<Completion> {
+    let parsed = parsed_module(db.upcast(), file);
+    identifiers(parsed.syntax().into())
+        .into_iter()
+        .map(|label| Completion { label })
+        .collect()
+}
+
+fn identifiers(node: AnyNodeRef) -> Vec<String> {
+    struct Visitor {
+        identifiers: Vec<String>,
+    }
+
+    impl<'a> SourceOrderVisitor<'a> for Visitor {
+        fn enter_node(&mut self, node: AnyNodeRef<'a>) -> TraversalSignal {
```

Comment:
Can we rely on `visit_identifier` here instead? Or, did you encounter any issues when using that method?

Reply:
Yeah! Just didn't know about it. Nice catch. :-) Much simpler.